### PR TITLE
internal: Deunwrap generate_derive

### PR DESCRIPTION
**Pull request by alibektas**
#15398 subtask 1. Since the editing closure has arms, I did something *experimental* ( in this case just a clever term for bad code ) to bypass creating an `Option` but I am ready to change this.

### CODE REVIEW

File: crates/ide-assists/src/handlers/generate_derive.rs [hunk 1]
```diff
@@ -27,33 +27,45 @@ pub(crate) fn generate_derive(acc: &mut Assists, ctx: &AssistContext<'_>) -> Opt
...
-                    make::token_tree(T!['('], vec![]).clone_for_update(),
-                ))
-                .clone_for_update();
-
                 let nominal = edit.make_mut(nominal);
                 nominal.add_attr(derive.clone());
 
-                edit.add_tabstop_before_token(
-                    cap,
-                    derive.meta().unwrap().token_tree().unwrap().r_paren_token().unwrap(),
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/generate_derive.rs:50]: Let's do an expect here, make should always create proper trees

**Review by Veykril — COMMENTED** [review]
Looking for the derive upfront is fine, but let's defer the `make` calls to the closure again

**Review by lnicola — APPROVED** [review]

**Pull request merged**